solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -229,7 +229,7 @@ impl<'a> FlashbotsSolutionSubmitter<'a> {
                         tracing::error!("flashbots cancellation failed: {:?}", err);
                     }
                 }
-                return err;
+                return Err(anyhow!("flashbots failed simulation: {}", err));
```

Comment:
Is Ok returned in any of the branches? Otherwise wouldn't it be just enough to change the error type?

Reply:
True, fixed.